Makefile

```diff
@@ -112,122 +217,103 @@ docker-build: generate-deepcopy build-for-docker .dockerflag.mk ## Build docker
-bin/ginkgo_v1:
-	GOBIN=$(PROJECT_DIR)/bin go install github.com/onsi/ginkgo/ginkgo@v1.16.5
-	mv $(PROJECT_DIR)/bin/ginkgo $(PROJECT_DIR)/bin/ginkgo_v1
-bin/mockgen:
-	GOBIN=$(PROJECT_DIR)/bin go install github.com/golang/mock/mockgen@v1.6.0
-bin/kustomize: ## Install kustomize to bin.
-	@mkdir -p bin
-	cd bin && curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash
-export K8S_VERSION=1.19.2
-bin/kubectl bin/kube-apiserver bin/etcd &:
-	curl --silent -L "https://go.kubebuilder.io/test-tools/${K8S_VERSION}/$(shell go env GOOS)/$(shell go env GOARCH)" --output - | \
-		tar -C ./ --strip-components=1 -zvxf -
-
-##@ Cleanup
-
-.PHONY: clean
-clean: ## Clean.
-	rm -rf $(RELEASE_DIR)
-	rm -rf bin
-	rm -rf cluster-api
-
-##@ Testing
-
-# Tell envtest to use local bins for etcd, kubeapi-server, and kubectl.
-export KUBEBUILDER_ASSETS=$(PROJECT_DIR)/bin
-
-.PHONY: test
-test: generate-mocks lint bin/ginkgo bin/kubectl bin/kube-apiserver bin/etcd ## Run tests. At the moment this is only unit tests.
-	@# The following is a slightly funky way to make sure the ginkgo statements are removed regardless the test results.
-	@ginkgo_v2 --label-filter="!integ" --cover -coverprofile cover.out --covermode=atomic -v ./api/... ./controllers/... ./pkg/...; EXIT_STATUS=$$?;\
-		./hack/testing_ginkgo_recover_statements.sh --remove; exit $$EXIT_STATUS
-
-.PHONY: generate-mocks
-generate-mocks: bin/mockgen generate-deepcopy pkg/mocks/mock_client.go $(shell find ./pkg/mocks -type f -name "mock*.go") ## Generate mocks needed for testing. Primarily mocks of the cloud package.
-pkg/mocks/mock%.go: $(shell find ./pkg/cloud -type f -name "*test*" -prune -o -print)
-	go generate ./...
-
 ##@ Tilt
+## --------------------------------------
+## Tilt Development
+## --------------------------------------
 
 .PHONY: tilt-up
-tilt-up: cluster-api kind-cluster cluster-api/tilt-settings.json manifests cloud-config ## Setup and run tilt for development.
+tilt-up: cluster-api kind-cluster cluster-api/tilt-settings.json generate-manifests cloud-config ## Setup and run tilt for development.
 	export CLOUDSTACK_B64ENCODED_SECRET=$$(base64 -w0 -i cloud-config 2>/dev/null || base64 -b 0 -i cloud-config) && cd cluster-api && tilt up
 
 .PHONY: kind-cluster
 kind-cluster: cluster-api ## Create a kind cluster with a local Docker repository.
 	-./cluster-api/hack/kind-install-for-capd.sh
 
 cluster-api: ## Clone cluster-api repository for tilt use.
-	git clone --branch v1.0.0 https://github.com/kubernetes-sigs/cluster-api.git
+	git clone --branch v1.0.0 --depth 1 https://github.com/kubernetes-sigs/cluster-api.git
 
 cluster-api/tilt-settings.json: hack/tilt-settings.json cluster-api
 	cp ./hack/tilt-settings.json cluster-api
 
-##@ End-to-End Testing
+##@ Tests
+## --------------------------------------
+## Tests
+## --------------------------------------
+
+export KUBEBUILDER_ASSETS=$(REPO_ROOT)/$(TOOLS_BIN_DIR)
```

Comment:
If we make `TOOLS_BIN_DIR` project root relative to begin with then this var can go away.

Reply:
Not necessarily, it requires the absolute path not a relative one